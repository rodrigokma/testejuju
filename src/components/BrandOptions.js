const BrandOptions = ({ value = "", onChange, brands }) => {
  return (
    <div className="brand-options">
      <select value={value} onChange={onChange}>
        <option data-testid="select-brand" value="">
          Select a brand
        </option>

        {brands &&
          brands.map((brand) => (
            <option
              data-testid="brand-option"
              key={`brand-${brand.codigo}`}
              value={brand.codigo}
            >
              {brand.nome}
            </option>
          ))}
      </select>
    </div>
  );
};

export default BrandOptions;
