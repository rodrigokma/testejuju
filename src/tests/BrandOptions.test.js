import React from "react";
import BrandOptions from "../components/BrandOptions";
import { render } from "@testing-library/react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import TestRenderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

describe("Brand options component tests", () => {
  const brandOptionsJsx = (
    <BrandOptions
      onChange={jest.fn()}
      brands={[
        { nome: "chevrolet", codigo: "93" },
        { nome: "nissan", codigo: "92" },
      ]}
    />
  );

  const component = shallow(brandOptionsJsx);

  test("renders component", () => {
    expect(component.exists()).toBe(true);
  });

  test("matches the snapshot", () => {
    const componentSnapshot = TestRenderer.create(brandOptionsJsx);
    expect(componentSnapshot).toMatchSnapshot();
  });

  test("has a 'Select a brand' as first option", () => {
    const { getByTestId } = render(brandOptionsJsx);

    expect(getByTestId("select-brand")).toHaveTextContent("Select a brand");
  });

  test("renders two brand options", () => {
    expect(component.find("option")).toHaveLength(3);
  });
});
