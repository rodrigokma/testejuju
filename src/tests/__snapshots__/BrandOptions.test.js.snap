// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand options component tests matches the snapshot 1`] = `
<div
  className="brand-options"
>
  <select
    onChange={[MockFunction]}
    value=""
  >
    <option
      data-testid="select-brand"
      value=""
    >
      Select a brand
    </option>
    <option
      data-testid="brand-option"
      value="93"
    >
      chevrolet
    </option>
    <option
      data-testid="brand-option"
      value="92"
    >
      nissan
    </option>
  </select>
</div>
`;
